src/System.Net.Http/src/System/Net/Headers/AuthenticationHeaderValue.cs

```diff
@@ -74,19 +74,19 @@ public override bool Equals(object obj)
 
             if (string.IsNullOrEmpty(_parameter) && string.IsNullOrEmpty(other._parameter))
             {
-                return (string.Compare(_scheme, other._scheme, StringComparison.OrdinalIgnoreCase) == 0);
+                return (string.Equals(_scheme, other._scheme, StringComparison.OrdinalIgnoreCase));
             }
             else
             {
                 // Since we can't parse the parameter, we use case-sensitive comparison.
-                return (string.Compare(_scheme, other._scheme, StringComparison.OrdinalIgnoreCase) == 0) &&
-                    (string.CompareOrdinal(_parameter, other._parameter) == 0);
+                return string.Equals(_scheme, other._scheme, StringComparison.OrdinalIgnoreCase) &&
+                    string.Equals(_parameter, other._parameter, StringComparison.Ordinal);
             }
         }
 
         public override int GetHashCode()
         {
-            int result = _scheme.ToLowerInvariant().GetHashCode();
+            int result = StringComparer.OrdinalIgnoreCase.GetHashCode(_scheme);
```

Comment:
Upon reading this:
http://stackoverflow.com/questions/11475737/gethashcode-for-ordinalignorecase-dependent-string-classes

and thinking about this:
"Two unequal objects can have the same hashcode. Though two equal objects should never have different hashcodes. If you use InvariantCulture for your hashcode it will still be correct as far as the contract for Equals goes if it's implemented in terms of OrdinalIgnoreCase."

I think the changes are fine.  If we were worried about serialization and having the hash codes changed like this, that might be a concern.  But the use case here is to support equality checks of the strong-type header classes.

So, bottom line, I'm ok w/ this.

System.Net.Http : LGTM :+1: